portfolio/src/main/java/com/google/sps/servlets/MarkerServlet.java

```diff
@@ -123,4 +149,79 @@ public Location getUserLocation(HttpServletRequest request){
     Location userLocation = new Location(lat, lng);
     return userLocation;
   }
+
+  private boolean validateMarker(double reportLat, double reportLng, String reportDate,
+   String reportTime, String reportCrime) {
+    Query query = new Query(ENTITY_TITLE);
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      double lat = (double) entity.getProperty("lat");
+      double lng = (double) entity.getProperty("lng");
+      String crime = (String) entity.getProperty("crimeType");
+      String date = (String) entity.getProperty("date");
+      String time = (String) entity.getProperty("time");
+
+      if (sameLocation(lat, lng, reportLat, reportLng) && sameDate(reportDate, date) 
```

Comment:
you could just do return sameLocation(...) && sameDate ...

Reply:
Done!